LDAR_Sim/src/ldar_sim_run.py

```diff
@@ -232,52 +248,38 @@ def simulate(
     )
     for simulation in range(simulation_count):
         print(f"......Simulating set {simulation}")
-        # read in pregen emissions
+        # read in pre-generated emissions
         infra = read_in_emissions(infrastructure, generator_dir, simulation)
         # -- Run simulations --
+        # -- Set up inputs for each program in a simulation set
         prog_data = []
-        prog_names = []
         for program in programs:
-            # TODO: get rid of state and split out into weather/daylight
-            prog_infra = copy.deepcopy(infra)
-            sites: list[Site] = prog_infra._sites
             meth_params = {}
             for meth in programs[program]["method_labels"]:
                 meth_params[meth] = methods[meth]
-            prog: Program = Program(
-                program,
-                weather,
-                daylight,
-                meth_params,
-                sites,
-                date(*virtual_world["start_date"]),
-                date(*virtual_world["end_date"]),
-                virtual_world["consider_weather"],
-            )
-            prog_names.append(program)
             prog_data.append(
                 (
+                    daylight,
+                    weather,
                     simulation,
-                    prog,
+                    program,
+                    meth_params,
                     sim_params,
                     virtual_world,
-                    prog_infra,
+                    infra,
                     in_dir,
                     out_dir,
                     seed_timeseries,
                 )
             )
-        for index, prog_tuple in enumerate(prog_data):
-            p_name = prog_names[index]
-            print(f".........Simulating program: {p_name}")
-            simulate(*prog_tuple)
-            print(f".........Finished simulating program: {p_name}")
-        # with mp.Pool(processes=sim_params["n_processes"]) as p:
-        #     sim_outputs = p.starmap(
-        #         simulate,
-        #         zip(*prog_data),
-        #     )
-        # print(f"Finished simulating set {simulation}")
+        with mp.Pool(processes=sim_params["n_processes"]) as p:
+            sim_outputs = p.starmap(
+                simulate,
+                prog_data,
+            )
+        gc.collect()
```

Comment:
Correction: I don't think this is actually inside the loop calling starmap on repeat

Reply:
The garbage collector is also called in the simulate function after the run_simulate() before each process ends. This is to clean up in between the different simulation sets.

Reply:
Sorry I was unclear, what I meant is that it looks like this is completely outside the loop running simulation sets and will only get called once all simulation sets are finished instead of in between simulation sets.